lte/gateway/c/core/oai/test/spgw_task/CMakeLists.txt

```diff
@@ -36,7 +36,9 @@ target_link_libraries(SPGW_TASK_TEST_LIB
 set_target_properties(SPGW_TASK_TEST_LIB PROPERTIES LINKER_LANGUAGE CXX)
 
 foreach (sgw_test spgw_state_converter spgw_procedures)
-  add_executable(${sgw_test}_test test_${sgw_test}.cpp)
-  target_link_libraries(${sgw_test}_test SPGW_TASK_TEST_LIB)
+  add_executable(${sgw_test}_test test_${sgw_test}.cpp
+              ${PROJECT_SOURCE_DIR}/tasks/grpc_service/grpc_service_task.c
```

Comment:
Is this line necessary? Is the file not included by linking `TASK_GRPC_SERVICE`?

Reply:
No, grpc_service_task.c is not included while generating TASK_GRPC_SERVICE. Even while generating mme binary, it's explicitly included. 
https://github.com/magma/magma/blob/master/lte/gateway/c/core/oai/oai_mme/CMakeLists.txt#L37

Reply:
Hmm that seems weird. Have you tried adding the file to the library? /workspaces/magma/lte/gateway/c/core/oai/tasks/grpc_service/CMakeLists.txt

Reply:
ok i was curious so tried it out :) This seems to fix the weird anti pattern https://github.com/magma/magma/pull/10324

Reply:
I followed the existing pattern.
Thanks @themarwhal Sure I will cherry-pick these changes.